### ISSUE TITLE
Option to set rounded corners?

### ISSUE THREAD

**Issue by SFR-git** [prikhi/lightdm-mini-greeter#63]
Would be really cool if you could make the corner round, and also be able to set the radius in px

**Comment by prikhi** [comment]
Yeah I'll check out adding something like `window-border-radius` & `password-border-radius`

**Comment by prikhi** [comment]
:thinking: password input is easy, I can just set `border-radius: ...` in the `#password` CSS, but when I try the same with the main window, I get black corners where the round occurs:

![2021-02-11-163722_342x115_scrot](https://user-images.githubusercontent.com/1304102/107701997-a33cfd00-6c87-11eb-8cf7-3fdfff1932cf.png)

**Comment by prikhi** [comment]
Hmm hmm, just gonna add in the `password-border-radius` for now while I figure out the main window - moving that discussion to #67.

**Comment by prikhi** [comment]
Tried for a while to get this working. I have a feeling it might be due to transparency & compositing, but not sure why it works for the password input.

I may have to re-jigger the GUI attachments so the main window is "inside" the background window, instead of "on top" of it...

**Comment by charlie45000** [comment]
I found a solution. I may setup a pull request when I have time:

```diff
diff --git a/src/ui.c b/src/ui.c
index 628f185..8b8e68e 100644
--- a/src/ui.c
+++ b/src/ui.c
@@ -169,6 +169,17 @@ static void setup_main_window(Config *config, UI *ui)
 
     gtk_container_set_border_width(GTK_CONTAINER(main_window), config->layout_spacing);
     gtk_widget_set_name(GTK_WIDGET(main_window), "main");
+    GdkScreen *screen = gtk_widget_get_screen(GTK_WIDGET(main_window));
+    GdkVisual *colormap = gdk_screen_get_rgba_visual(screen);
+
+    if (!colormap)
+    {
+        g_warning("Your screen does not support alpha channels!");
+    }
+    else
+    {
+        gtk_widget_set_visual(GTK_WIDGET(main_window), colormap);
+    }
 
     g_signal_connect(main_window, "show", G_CALLBACK(place_main_window), NULL);
     g_signal_connect(main_window, "realize", G_CALLBACK(hide_mouse_cursor), NULL);
```

This requires to have a compositor running though. I put this in the lightdm.conf: `greeter-setup-script=/usr/bin/picom -b`

The result is (with no password label text and rgba colors for password wondow and text field):

<img width="470" height="227" alt="Image" src="https://github.com/user-attachments/assets/ef2d5197-3c1c-4e3c-95bf-610b7fe92016" />
